Validation/HGCalValidation/test/HGCalWaferHitCheck.cc

```diff
@@ -0,0 +1,262 @@
+// system include files
+#include <cmath>
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "DataFormats/ForwardDetId/interface/ForwardSubdetector.h"
+#include "DataFormats/ForwardDetId/interface/HFNoseDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCScintillatorDetId.h"
+#include "DataFormats/HGCDigi/interface/HGCDigiCollections.h"
+#include "DataFormats/HGCRecHit/interface/HGCRecHitCollections.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDAnalyzer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/HGCalCommonData/interface/HGCalDDDConstants.h"
+#include "Geometry/HGCalCommonData/interface/HGCalWaferIndex.h"
+
+#include "SimDataFormats/CaloHit/interface/PCaloHitContainer.h"
+#include "SimDataFormats/CaloHit/interface/PCaloHit.h"
+
+namespace HGCalValidSimhitCheck {
+  struct Counters {
+    Counters() {
+      badTypes_.clear();
+      occupancy_.clear();
+      goodChannels_.clear();
+    }
+    mutable std::map<int, int> badTypes_, occupancy_;
+    mutable std::vector<int> goodChannels_;
```

Comment:
Following will silence static analyzer warnings, and make it more clear that the access to these objects is protected by `mtx_`.
```suggestion
    CMS_THREAD_GUARD(mtx_) mutable std::map<int, int> badTypes_, occupancy_;
    CMS_THREAD_GUARD(mtx_) mutable std::vector<int> goodChannels_;
```